pkg/azurediskplugin/main.go

```diff
@@ -80,44 +87,153 @@ func main() {
 		klog.Warning("nodeid is empty")
 	}
 
+	getDriverConfig()
 	exportMetrics()
 	handle()
 	os.Exit(0)
 }
 
-func handle() {
-	driverOptions := azuredisk.DriverOptions{
-		NodeID:                     *nodeID,
-		DriverName:                 *driverName,
-		VolumeAttachLimit:          *volumeAttachLimit,
-		EnablePerfOptimization:     *enablePerfOptimization,
-		CloudConfigSecretName:      *cloudConfigSecretName,
-		CloudConfigSecretNamespace: *cloudConfigSecretNamespace,
-		CustomUserAgent:            *customUserAgent,
-		UserAgentSuffix:            *userAgentSuffix,
-		UseCSIProxyGAInterface:     *useCSIProxyGAInterface,
-		EnableDiskOnlineResize:     *enableDiskOnlineResize,
-		AllowEmptyCloudConfig:      *allowEmptyCloudConfig,
-		EnableAsyncAttach:          *enableAsyncAttach,
-		EnableListVolumes:          *enableListVolumes,
-		EnableListSnapshots:        *enableListSnapshots,
-		SupportZone:                *supportZone,
-		GetNodeInfoFromLabels:      *getNodeInfoFromLabels,
-		EnableDiskCapacityCheck:    *enableDiskCapacityCheck,
-		VMSSCacheTTLInSeconds:      *vmssCacheTTLInSeconds,
-		VMType:                     *vmType,
-		RestClientQPS:              *kubeClientQPS,
+func getDriverConfig() {
+	if *driverConfigPath != "" {
+		// Read config file and convert to a driveConfig object
+		yamlFile, err := ioutil.ReadFile(*driverConfigPath)
+		if err != nil {
+			klog.Fatalf("failed to get the driver config, error: %v", err)
+		}
+
+		err = yaml.Unmarshal(yamlFile, &driverConfig)
```

Comment:
Hmmm. I wonder if we should take the [same approach as `kube-scheduler` for reading its configuration](https://github.com/kubernetes/kubernetes/blob/d2cea9475bb7d2c2a1fa516ddce4af4f0cf3a4ef/cmd/kube-scheduler/app/options/configfile.go#L44) rather than using `yaml.Unmarshal` directly?

Reply:
I'll take a look at it.